File: regexlib/python_re2_test_file/regexlib_1087.py
# 1087
# ^[^']*?\&lt;\s*Assembly\s*:\s*AssemblyVersion\s*\(\s*&quot;(\*|[0-9]+.\*|[0-9]+.[0-9]+.\*|[0-9]+.[0-9]+.[0-9]+.\*|[0-9]+.[0-9]+.[0-9]+.[0-9]+)&quot;\s*\)\s*\&gt;.*$
# EXPONENT
# nums:5
# EXPONENT AttackString:"*01*0101*010101*&lt;Assembly:AssemblyVersion(&quot;"+"0101"*64+"!1 _SLQ_2"

import re2 as re
from time import perf_counter

regex = """^[^']*?\&lt;\s*Assembly\s*:\s*AssemblyVersion\s*\(\s*&quot;(\*|[0-9]+.\*|[0-9]+.[0-9]+.\*|[0-9]+.[0-9]+.[0-9]+.\*|[0-9]+.[0-9]+.[0-9]+.[0-9]+)&quot;\s*\)\s*\&gt;.*$"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "*01*0101*010101*&lt;Assembly:AssemblyVersion(&quot;" + "0101" * i * 1 + "!1 _SLQ_2"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *1}: took {DURATION} seconds!")

File: regexlib/javascript_test_file/regexlib_1087.js
/* 1087
 * ^[^']*?\&lt;\s*Assembly\s*:\s*AssemblyVersion\s*\(\s*&quot;(\*|[0-9]+.\*|[0-9]+.[0-9]+.\*|[0-9]+.[0-9]+.[0-9]+.\*|[0-9]+.[0-9]+.[0-9]+.[0-9]+)&quot;\s*\)\s*\&gt;.*$
 * EXPONENT
 * nums:3
 * EXPONENT AttackString:"*01*0101*010101*&lt;Assembly:AssemblyVersion(&quot;"+"0101"*64+"!1 _SLQ_2"
 */
var REGEX = /^[^']*?\&lt;\s*Assembly\s*:\s*AssemblyVersion\s*\(\s*&quot;(\*|[0-9]+.\*|[0-9]+.[0-9]+.\*|[0-9]+.[0-9]+.[0-9]+.\*|[0-9]+.[0-9]+.[0-9]+.[0-9]+)&quot;\s*\)\s*\&gt;.*$/
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '*01*0101*010101*&lt;Assembly:AssemblyVersion(&quot;' + '0101'.repeat(i*1) + '!1 _SLQ_2'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 1 + ": " + time_cost+" ms")
}